### PR TITLE
Support constraint in plot slice

### PR DESCRIPTION
**Pull request by hrntsm**
## Motivation

#4833

Make plot_slice plots support optimization with constraints

## Description of the changes

The results of optimization with the following objective function are displayed in slices.

```py
def objective(trial):
    x = trial.suggest_float("x", -15, 30)
    y = trial.suggest_float("y", -15, 30)
    v0 = 4 * x**2 + 4 * y**2
    trial.set_user_attr("constraint", [1000 - v0])
    return v0
```

this pr
newplot (5)

without constraint plot
newplot (4)

### CODE REVIEW

**Comment by hrntsm** [comment]
matplotlib plot result
Screenshot 2023-09-09 at 17 55 45

**Comment by c-bata** [comment]
@not522 @eukaryo Could you review this PR?

File: optuna/visualization/matplotlib/_slice.py [hunk 1]
```diff
@@ -129,35 +130,61 @@ def _generate_slice_subplot(
...
-        for x_label in subplot_info.x_labels:
-            for y, number in points_dict[x_label]:
-                x_values.append(str(x_label))
-                y_values.append(y)
-                c_values.append(number)
+        feasible_x, feasible_y, feasible_c = _get_categorical_plot_values(subplot_info, feasible)
+        infeasible_x, infeasible_y, _ = _get_categorical_plot_values(subplot_info, infeasible)
         scale = "categorical"
-    xlim = _calc_lim_with_padding(x_values, padding_ratio, scale)
+    xlim = _calc_lim_with_padding(feasible_x, padding_ratio, scale)
```
Comment (not522) [inline optuna/visualization/matplotlib/_slice.py:161]: Infeasible trials may be outside of feasible trials.

```suggestion
    xlim = _calc_lim_with_padding(feasible_x + infeasible_x, padding_ratio, scale)
```

**Review by not522 — COMMENTED** [review]
Thank you for your PR! I added a comment, please check it.

**Review by not522 — APPROVED** [review]
LGTM!

**Comment by eukaryo** [comment]
LGTM!

**Pull request merged**